src/backend-c/runtime/tests/cast_queue.c

```diff
@@ -0,0 +1,80 @@
+#include <assert.h>
+#include "cast_queue.h"
+
+void dequeue_1(){
+  cast_queue* q = allocate_cast_queue();
+  assert(q->capacity == 0);
+  assert(q->front == 0);
+  assert(q->rear == 0);
+  grift_obj addr, type;
+  ENQUEUE(q, addr, type, 991, 5);
+  assert(q->capacity == CAST_QUEUE_INIT_CAPACITY);
+  suspended_cast sc = cast_queue_dequeue(q);
+  assert(addr.as_int == sc.address.as_int);
+  assert(type.as_int == sc.type.as_int);
+}
+
+void dequeue_2(){
+  cast_queue* q = allocate_cast_queue();
+  grift_obj addr1, type1;
+  ENQUEUE(q, addr1, type1, 991, 5);
+  assert(q->front == 0 && q->rear == 1);
+  grift_obj addr2, type2;
+  ENQUEUE(q, addr2, type2, 892, 8);
+  assert(q->front == 0 && q->rear == 2);
+  suspended_cast sc = cast_queue_dequeue(q);
+  assert(q->front == 1 && q->rear == 2);
+  assert(addr1.as_int == sc.address.as_int);
+  assert(type1.as_int == sc.type.as_int);
+  sc = cast_queue_dequeue(q);
+  assert(q->front == 2 && q->rear == 2);
+  assert(addr2.as_int == sc.address.as_int);
+  assert(type2.as_int == sc.type.as_int);
+}
+
+void dequeue_3(){
+  cast_queue* q = allocate_cast_queue();
+  grift_obj addr1, type1;
+  ENQUEUE(q, addr1, type1, 991, 5);
+  suspended_cast sc = cast_queue_dequeue(q);
+  assert(q->front == 1 && q->rear == 1);
+  assert(addr1.as_int == sc.address.as_int);
+  grift_obj addr2, type2;
+  ENQUEUE(q, addr2, type2, 892, 8);
+  sc = cast_queue_dequeue(q);
+  assert(q->front == 2 && q->rear == 2);
```

Comment:
You might not want to check the internal state of the queue so directly. Instead you could add a `is_valid_state` predicate that both describes and succinctly checks the validity of the state. This way if you change the design you won't have to flop bits in each test case.